cluster-autoscaler/cloudprovider/gce/templates_test.go

```diff
@@ -186,6 +187,23 @@ func TestBuildNodeFromTemplateSetsResources(t *testing.T) {
 			attachedLocalSSDCount:         4,
 			expectedErr:                   false,
 		},
+		{
+			scenario:                      "extended_resources present in kube-env",
+			kubeEnv:                       "AUTOSCALER_ENV_VARS: kube_reserved=cpu=0,memory=0,ephemeral-storage=10Gi;os_distribution=cos;os=linux;ephemeral_storage_local_ssd_count=2;extended_resources=someResource=2,anotherResource=1G\n",
```

Comment:
Please also add some tests for handling malformed `extended_resources` (e.g. `extended_resources=foo`, `extended_resources=foo=bar`).

Reply:
Sound good 👍 

I've added a test case to verify that malformed extended resources won't break the template building function, as well as more test case around the extraction function itself.